src/lightning/pytorch/core/module.py

```diff
@@ -648,21 +650,32 @@ def forward(self, *args: Any, **kwargs: Any) -> Any:
         """
         return super().forward(*args, **kwargs)
 
-    def training_step(self, *args: Any, **kwargs: Any) -> STEP_OUTPUT:  # type: ignore[return-value]
+    @overload
+    def training_step(self, batch: Any, batch_idx: int) -> Optional[STEP_OUTPUT]:
+        ...
+
+    @overload
+    def training_step(self, batch: Any, batch_idx: int, dataloader_idx: int) -> Optional[STEP_OUTPUT]:
+        ...
+
+    @overload
+    def training_step(self, dataloader_iter: Iterable[DataLoader[Any]]) -> Optional[STEP_OUTPUT]:
+        ...
+
+    def training_step(self, *args: Any, **kwargs: Any) -> Optional[STEP_OUTPUT]:
         r"""Here you compute and return the training loss and some additional metrics for e.g. the progress bar or
         logger.
 
         Args:
-            batch (:class:`~torch.Tensor` | (:class:`~torch.Tensor`, ...) | [:class:`~torch.Tensor`, ...]):
-                The output of your :class:`~torch.utils.data.DataLoader`. A tensor, tuple or list.
-            batch_idx (``int``): Integer displaying index of this batch
+            batch: The output of your :class:`~torch.utils.data.DataLoader`.
+            batch_idx: The index of this batch.
+            dataloader_idx: The index of the dataloader that produced this batch.
+                (only if multiple dataloaders used)
 
         Return:
-            Any of.
-
             - :class:`~torch.Tensor` - The loss tensor
-            - ``dict`` - A dictionary. Can include any keys, but must include the key ``'loss'``
-            - ``None`` - Training will skip to the next batch. This is only for automatic optimization.
+            - ``dict`` - A dictionary. Can include any keys, but must include the key ``'loss'``.
+            - ``None`` - Skip to the next batch. This is only for automatic optimization.
```

Comment:
```suggestion
            - ``None`` - Skip to the next batch. This is only supported for automatic optimization.
```
(while we are at it)